### ISSUE TITLE
Helper functions for choosing data types for bias and weight in Flux chains?

### ISSUE THREAD

**Issue by B-LIE** [FluxML/Flux.jl#1548]
I know many are happy with Float32, but others training Neural Networks on CPUs will wonder how to choose Float64.

I've heard there are helper functions to set the data type of bias and weight.
* Would it be possible to document these in the documentation?
* If `md` is my model, could it be done as simply as `Float64(md)` or `convert(Float64, md)` -- and similarly for other types?

**Comment by DhairyaLGandhi** [comment]
`Flux.f64` is your friend. Would you be up to write simple docstrings for it?

**Comment by DhairyaLGandhi** [comment]
`f64(md)` would do it.

**Comment by B-LIE** [comment]
Sure, I can do it... but I've never done such before. Where do I insert it? (I can probably figure out what a docstring is... I should say that I started my programming practice on an Altair computer with 4kB RAM in high school in 1976, and then with Fortran 4 at university, so I'm not super good with programming...)

One thing: by testing, I found that `f64(md)` only works when I *create* the model, i.e., `md = Chain(....) |> f64` works, but if I applied `f64` in post, after `md` had been created, I didn't see a change in the parameter type.

**Comment by DhairyaLGandhi** [comment]
It works the same as `gpu`, so it picks up the fields as described by `@functor` and so on.

To insert the docstrings, you prolly want to do that here https://github.com/FluxML/Flux.jl/blob/5c407168012b696913ec79afd1c995235afd36f1/src/functor.jl#L77

The way to do that is simply have a multiline comment sitting on top of the function definitions (see the rest of the script for some quick helpers for the format)

Happy to help further too!

**Comment by DhairyaLGandhi** [comment]
It should be possible to call the function whenever :)

**Comment by B-LIE** [comment]
OK -- just so that I don't mess up things... should I
* fork out (or whatever it is called) a copy on my computer
* insert the docstrings
* submit it for review (or whatever it is called)
I can do that, but I don't want to mess things up -- so I should have some certainty in that I do things right

**Comment by darsnack** [comment]
You can't directly write to the Flux main repo (only your fork). So it isn't possible to mess things up. Give it a shot and we can help you where you get stuck.

**Comment by darsnack** [comment]
I explain Github contributions to another first time contributor [here](https://julialang.zulipchat.com/#narrow/stream/237432-ml-ecosystem-coordination/topic/FastAI.2Ejl.20issues) towards the end of the thread. Maybe it will help you too?

**Comment by B-LIE** [comment]
Thanks. I'll take a look tomorrow. It's 1 a.m here, now :-)

Get Outlook for Android<https://aka.ms/AAb9ysg>

________________________________

**Comment by B-LIE** [comment]
OK... I managed to find the src director, etc. I did the following:
1. Forked the code
2. Entered the correct subdirectory.
3. Chose "edit", and edited the code
4. At the bottom of the window, I "committed" the changes.

From a previous case where a colleague "held me by the hand" why I did a change, I seem to recall that this "commit" has only changed the code on my private computer (or: is it on my private GitHub account??), and that I need to "push" it (or something) to upload it to the public repository.

But I didn't find any buttons or menu choices to "push" the changes.
